micromamba/src/run.cpp

```diff
@@ -100,16 +285,45 @@ set_run_command(CLI::App* subcom)
             exit(1);
         }
 
+        LOG_DEBUG << "Currently running processes: " << get_all_running_processes_info();
+        LOG_DEBUG << "Remaining args to run as command: " << join(" ", command); // TODO: lower log level and then check why they never print in info and debug
+
         // replace the wrapping bash with new process entirely
-        #ifndef _WIN32
+#ifndef _WIN32
         if (command.front() != "exec")
             command.insert(command.begin(), "exec");
-        #endif
+
+        // Lock the process directory to read and write in it until we manage to run the file or exit.
+        static auto proc_dir_lock = lock_proc_dir();  // Note: this object is static only so that calls to `std::exit()` will call the destructor.
+
+        const std::string process_name = [&]{
+            // Insert a unique process name associated to the command, either specified by the user or generated.
+            command.reserve(4); // We need at least 4 objects to not move around.
+
+            const auto exe_name_it = std::next(command.begin());
+            if(specific_process_name.empty())
+            {
+                const auto unique_name = generate_unique_process_name(*exe_name_it);
+                command.insert(exe_name_it, { {"-a"}, unique_name });
+                return unique_name;
+            }
+            else
+            {
+                if(is_process_name_running(specific_process_name))
+                {
+                    LOG_ERROR << fmt::format("Another process with name '{}' is currently running.", specific_process_name);
```

Comment:
maybe if you throw a `std::runtime_error` here you don't need the static lockfile? Since later you catch the error.

Reply:
Yes that would work, I'll change that.

There are other exit calls after we finish using the lock (after line 391 basically) but that shouldnt impact the present code.

I'm considering replacing all the exit calls in this lambda by throws except the successful one to help future modifications to not rely on exit and break RAII usage here. I could then move the whole code in an inner scope and the last successful exit call out of that scope, that would basically remove the statics in this lambda - but it might make the code look weird haha